src/Api/Serializer/ContextBuilder/AbstractAwareContextBuilder.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CommerceWeavers\SyliusTpayPlugin\Api\Serializer\ContextBuilder;
+
+use ApiPlatform\Serializer\SerializerContextBuilderInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Serializer\Normalizer\AbstractNormalizer;
+
+abstract class AbstractAwareContextBuilder implements AwareContextBuilderInterface
+{
+    public function __construct(
+        private readonly SerializerContextBuilderInterface $decoratedContextBuilder,
+    ) {
+    }
+
+    public function createFromRequest(Request $request, bool $normalization, array $extractedAttributes = null): array
+    {
+        $context = $this->decoratedContextBuilder->createFromRequest($request, $normalization, $extractedAttributes);
+
+        if (!$this->supports($request, $context, $extractedAttributes)) {
+            return $context;
+        }
+
+        /** @var string $inputClass */
+        $inputClass = $this->getInputClassFrom($context);
+        $constructorArgumentName = $this->getConstructorArgumentName($inputClass);
+
+        $context[AbstractNormalizer::DEFAULT_CONSTRUCTOR_ARGUMENTS][$inputClass] = array_merge(
+            $context[AbstractNormalizer::DEFAULT_CONSTRUCTOR_ARGUMENTS][$inputClass] ?? [],
+            [$constructorArgumentName => $request->attributes->get($this->getAttributeKey())],
+        );
+
+        return $context;
+    }
+
+    public function supports(Request $request, array $context, ?array $extractedAttributes): bool
+    {
+        $inputClass = $this->getInputClassFrom($context);
+
+        if (null === $inputClass) {
+            return false;
+        }
+
+        return is_a($inputClass, $this->getSupportedInterface(), true);
+    }
+
+    private function getInputClassFrom(array $context): ?string
+    {
+        return $context['input']['class'] ?? null;
+    }
+
+    private function getConstructorArgumentName(string $inputClass): string
+    {
+        if (!is_a($inputClass, $this->getSupportedInterface(), true)) {
```

Comment:
Using Webmozart Assert could be nicer:
```suggestion
        // if vars would be classes
        Assert::implementsInterface($inputClass, $this->getSupportedInterface(), 'message');
        // if they will be strings
        Assert::same($inputClass, $this->getSupportedInterface(), 'message');
```

But it is just preference and don't block anything IMO